src/Illuminate/Http/Client/Response.php

```diff
@@ -50,15 +50,21 @@ public function body()
     /**
      * Get the JSON decoded body of the response as an array or scalar value.
      *
+     * @param  string|null  $key
+     * @param  mixed  $default
      * @return mixed
      */
-    public function json()
+    public function json($key = null, $default = null)
```

Comment:
This is a breaking change to the method signature unfortunately.

Reply:
how is this a breaking change if both params are nullable?

Reply:
Because a package could extend that class and it would break with these changes:
```php
class A{public function json($key = null, $default = null){}}
class B extends A{public function json(){}}
```
results in:
`ErrorException: Declaration of B::json() should be compatible with A::json($key = NULL, $default = NULL)`

Reply:
you're right... forgot about that